Added code snippet for adding keys to Info.plist

theindianappguy opened the pull request:
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

theindianappguy commented:
coming from the android background I don't have much understanding of Info.plist hence I was not able to understand what exactly I have to do searched on the web not able to find the right solution  

It's quite simple but with a sample code it becomes quite easier to understand even for a newbie

theindianappguy commented:
@googlebot I signed it!

theindianappguy commented:
as I have only updated the Readme I have not updated the CHANGELOG.md please let me know if required

cyanglaz commented:
@theindianappguy Thank you for the PR.
Could you explain more in detail about what the README.md update is about? I find myself not able to get it :) 
And yes, REAMDE changes also requires version updates.

theindianappguy commented:
@cyanglaz I found it a bit hard to understand how to implement in IOS, so added screenshot so that people with no Experience with Native IOS app development can also add it easily

Also major of the developer use windows so usually not able to test on the ios device so added a screenshot of how the description will be reflected

cyanglaz commented:
@theindianappguy Ah, thanks! I wonder if we can just add a link to https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/AboutInformationPropertyListFiles.html which has the complete steps

theindianappguy commented:
@cyanglaz  yes we can, and if we want the image will be enough

cyanglaz commented:
@theindianappguy Let's link it to Apple's doc so we don't have to maintain this screenshot if somehow XCode's UI changes. WDYT? :)

theindianappguy commented:
I think the structure is not going to change if it will we will update the screenshot, the thing is I personally find it very easy to understand through the screenshot compared to going through a website to understand it all 

so I will recommend having screenshot which is basically what they have to do if there is any change, later on, we can update

theindianappguy commented:
@ShivamGoyal1899 being a ios developer it may seem to easy but since i am android developer i think it's important to have this, it will save people time and make it easy to implement

Thanks for code recommendation i have updated the pull request rather than having an image now i have added text which user can easily copy as well @cyanglaz please have a look and update

as we also have in camera i think it's a good add
<img width="502" alt="Screenshot 2020-03-15 at 10 00 31 AM" src="https://user-images.githubusercontent.com/55942632/76695313-d35cd400-66a3-11ea-9939-c7a62ef2b90c.png">

theindianappguy commented:
@ShivamGoyal1899 please check the camera plugin readme [here](https://github.com/flutter/plugins/tree/master/packages/camera), and if you recommend we should change the or text format add the key what would you recommend

stuartmorgan commented on packages/image_picker/README.md:
```diff
@@ -17,6 +17,17 @@ Add the following keys to your _Info.plist_ file, located in `<project root>/ios
 * `NSCameraUsageDescription` - describe why your app needs access to the camera. This is called _Privacy - Camera Usage Description_ in the visual editor.
 * `NSMicrophoneUsageDescription` - describe why your app needs access to the microphone, if you intend to record videos. This is called _Privacy - Microphone Usage Description_ in the visual editor.
```
Since there's a text version below now, these should use the XCode UI labels rather than the raw keys (as in the Camera README you referenced), and the wording should change accordingly (e.g., "Add the following rows" rather than "Add the following keys").

stuartmorgan commented on packages/image_picker/README.md:
```diff
@@ -17,6 +17,17 @@ Add the following keys to your _Info.plist_ file, located in `<project root>/ios
 * `NSCameraUsageDescription` - describe why your app needs access to the camera. This is called _Privacy - Camera Usage Description_ in the visual editor.
 * `NSMicrophoneUsageDescription` - describe why your app needs access to the microphone, if you intend to record videos. This is called _Privacy - Microphone Usage Description_ in the visual editor.
 
+Or in text format add the key:
```
"Or if you are editing the file as plain text, add:"

stuartmorgan commented on packages/image_picker/README.md:
````diff
@@ -17,6 +17,17 @@ Add the following keys to your _Info.plist_ file, located in `<project root>/ios
+
+``` xml
+<key>NSPhotoLibraryUsageDescription</key>
+<string>Used to demonstrate image picker plugin</string>
````
I'm concerned that these examples would lead people to write poor explanations, since they are very vague. How about "Insert your explanation for using the photo library here", and similarly below.

stuartmorgan reviewed (changes requested):
We apologize for the long delay in triaging this PR. We’re in the process of overhauling our PR triage system to respond much more quickly, as well as working through the backlog.

I think it's reasonable to add the actual text version to make it easier for developers encountering plists for the first time. Some minor comments, but otherwise I don't see any reason not to add this.

stuartmorgan commented:
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!
